plugins/apache_druid_logs.yaml

```
@@ -0,0 +1,217 @@
+    type: "string"
+    required: true
+  - name: enable_broker_logs
+    description: Enable to collect Apache Druid broker logs
+    type: bool
+    default: true
+  - name: broker_log_filename
+    description: Filename containing Apache druid broker logs
+    type: "string"
+    default: "broker.log"
+  - name: enable_coordinator_overlord_logs
+    description: Enable to collect Apache Druid coordinator and overlord logs
+    type: bool
+    default: true
+  - name: coordinator_overlord_log_filename
+    description: Filename containing Apache Druid coordinator and overlord logs
+    type: "string"
+    default: "coordinator-overlord.log"
+  - name: enable_historical_logs
+    description: Enable to collect Apache Druid historical logs
+    type: bool
+    default: true
+  - name: historical_log_filename
+    description: Filename containing Apache Druid historical logs
+    type: "string"
+    default: "historical.log"
+  - name: enable_middle_manager_logs
+    description: Enable to collect Apache Druid middle manager logs
+    type: bool
+    default: true
+  - name: middle_manager_log_filename
+    description: Filename containing Apache Druid middle manager logs
+    type: "string"
+    default: "middleManager.log"
+  - name: enable_router_logs
+    description: Enable to collect Apache Druid router logs
+    type: bool
+    default: true
+  - name: router_log_filename
+    description: Filename containing Apache Druid router logs
+    type: "string"
+    default: "router.log"
+  - name: enable_zookeeper_logs
+    description: Enable to collect Apache Druid ZooKeeper logs
+    type: bool
+    default: true
+  - name: zookeeper_log_filename
+    description: Filename containing Apache Druid ZooKeeper logs
+    type: "string"
+    default: "zookeeper.log"
+  - name: start_at
+    description: At startup, where to start reading logs from the file (`beginning` or `end`)
+    type: string
+    valid_values:
+     - beginning
+     - end
+    default: end
+  # Apache Druid logs do not include a timezone in their timestamps, so we need to set it explicitly.
+  - name: timezone
+    description: Timezone to use when parsing the timestamp
+    type: timezone
+    default: UTC
+template: |
+  receivers:
+    {{ if .enable_broker_logs }}
+    filelog/broker_logs:
+      include:
+        - '{{ .log_directory }}{{ .broker_log_filename }}'
+      multiline:
+        line_start_pattern: '\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2},\d{3}'
+      start_at: {{ .start_at }}
+      attributes:
+        log_type: 'druid.broker'
+      operators:
+        - type: regex_parser
+          regex: '(?P<timestamp>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2},\d{3})\s+(?P<severity>[A-Z]+)\s+(?P<message>\[(?P<source>[\S\s]+)\]\s+(?P<javaClass>[A-Za-z0-9\.$]+)\s+-\s+?[\S\s]+)'
+          severity:
+            parse_from: attributes.severity
+          timestamp:
+            parse_from: attributes.timestamp
+            layout: '%FT%T,%L'
+            location: {{ .timezone }}
```

Generally we set `parse_to: body` on all parsers, unless we have a good reason not to.

I've not tested this suggested change, but it should work and should apply to all parsers in this plugin.
```suggestion
        - type: regex_parser
          regex: '(?P<timestamp>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2},\d{3})\s+(?P<severity>[A-Z]+)\s+(?P<message>\[(?P<source>[\S\s]+)\]\s+(?P<javaClass>[A-Za-z0-9\.$]+)\s+-\s+?[\S\s]+)'
          parse_to: body
          severity:
            parse_from: body.severity
          timestamp:
            parse_from: body.timestamp
            layout: '%FT%T,%L'
            location: {{ .timezone }}
```

---

Gave this a try and it threw a bunch of errors up in my console. Here's one of the repeated errors:

`{"level":"error","ts":"2023-04-18T15:01:38.186-0400","caller":"helper/transformer.go:110","msg":"Failed to process entry","kind":"receiver","name":"plugin","data_type":"logs","kind":"receiver","name":"filelog/broker_logs","data_type":"logs","operator_id":"regex_parser","operator_type":"regex_parser","error":{"description":"time parser: log entry does not have the expected parse_from field","suggestion":"ensure that all entries forwarded to this parser contain the parse_from field","details":{"parse_from":"attributes.timestamp"}},"action":"send","entry":{"observed_timestamp":"2023-04-18T15:01:38.185776-04:00","timestamp":"0001-01-01T00:00:00Z","body":{"javaClass":"org.apache.druid.java.util.emitter.core.HttpPostEmitter","message":"[HttpPostEmitter-1] org.apache.druid.java.util.emitter.core.HttpPostEmitter - Failed to send events to url[http://localhost:8080]\norg.apache.druid.java.util.common.ISE: Emissions of events not successful[405: Method Not Allowed], with message[<html>\n<head>\n<meta http-equiv=\"Content-Type\" content=\"text/html;charset=ISO-8859-1\"/>\n<title>Error 405 HTTP method POST is not supported by this URL</title>\n</head>\n<body><h2>HTTP ERROR 405 HTTP method POST is not supported by this URL</h2>\n<table>\n<tr><th>URI:</th><td>/</td></tr>\n<tr><th>STATUS:</th><td>405</td></tr>\n<tr><th>MESSAGE:</th><td>HTTP method POST is not supported by this URL</td></tr>\n<tr><th>SERVLET:</th><td>org.eclipse.jetty.servlet.ServletHandler$Default404Servlet-7fcbe147</td></tr>\n</table>\n<hr/><a href=\"https://eclipse.org/jetty\">Powered by Jetty:// 9.4.48.v20220622</a><hr/>\n\n</body>\n</html>].\n\tat org.apache.druid.java.util.emitter.core.HttpPostEmitter$EmittingThread.send(HttpPostEmitter.java:813) ~[druid-core-25.0.0.jar:25.0.0]\n\tat org.apache.druid.java.util.emitter.core.HttpPostEmitter$EmittingThread.access$1900(HttpPostEmitter.java:464) ~[druid-core-25.0.0.jar:25.0.0]\n\tat org.apache.druid.java.util.emitter.core.HttpPostEmitter$EmittingThread$1.perform(HttpPostEmitter.java:667) ~[druid-core-25.0.0.jar:25.0.0]\n\tat org.apache.druid.java.util.emitter.core.HttpPostEmitter$EmittingThread$1.perform(HttpPostEmitter.java:663) ~[druid-core-25.0.0.jar:25.0.0]\n\tat org.apache.druid.java.util.common.RetryUtils.retry(RetryUtils.java:129) ~[druid-core-25.0.0.jar:25.0.0]\n\tat org.apache.druid.java.util.common.RetryUtils.retry(RetryUtils.java:81) ~[druid-core-25.0.0.jar:25.0.0]\n\tat org.apache.druid.java.util.common.RetryUtils.retry(RetryUtils.java:163) ~[druid-core-25.0.0.jar:25.0.0]\n\tat org.apache.druid.java.util.common.RetryUtils.retry(RetryUtils.java:153) ~[druid-core-25.0.0.jar:25.0.0]\n\tat org.apache.druid.java.util.emitter.core.HttpPostEmitter$EmittingThread.sendWithRetries(HttpPostEmitter.java:661) ~[druid-core-25.0.0.jar:25.0.0]\n\tat org.apache.druid.java.util.emitter.core.HttpPostEmitter$EmittingThread.tryEmitOneFailedBuffer(HttpPostEmitter.java:634) ~[druid-core-25.0.0.jar:25.0.0]\n\tat org.apache.druid.java.util.emitter.core.HttpPostEmitter$EmittingThread.run(HttpPostEmitter.java:504) ~[druid-core-25.0.0.jar:25.0.0]","severity":"ERROR","source":"HttpPostEmitter-1","timestamp":"2023-04-18T19:01:38,054"},"attributes":{"log.file.name":"broker.log","log_type":"druid.broker"},"severity":0,"scope_name":""},"stacktrace":"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper.(*TransformerOperator).HandleEntryError\n\t/Users/ian/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.75.0/operator/helper/transformer.go:110\ngithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper.(*ParserOperator).ParseWith\n\t/Users/ian/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.75.0/operator/helper/parser.go:186\ngithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper.(*ParserOperator).ProcessWithCallback\n\t/Users/ian/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.75.0/operator/helper/parser.go:123\ngithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper.(*ParserOperator).ProcessWith\n\t/Users/ian/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.75.0/operator/helper/parser.go:109\ngithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/parser/regex.(*Parser).Process\n\t/Users/ian/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.75.0/operator/parser/regex/regex.go:110\ngithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper.(*WriterOperator).Write\n\t/Users/ian/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.75.0/operator/helper/writer.go:64\ngithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/input/file.(*Input).emit\n\t/Users/ian/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.75.0/operator/input/file/file.go:65\ngithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Reader).ReadToEnd\n\t/Users/ian/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.75.0/fileconsumer/reader.go:102\ngithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.(*Manager).consume.func1\n\t/Users/ian/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.75.0/fileconsumer/file.go:151"}`

Am I missing something?

---

Did you set parse_to and update the severity / timestamp parse_from values?

```
- type: regex_parser
  regex:
  parse_to: body
```

parse_to defaults to attributes. https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/docs/operators/regex_parser.md

---

Just to chime in here we are likely going to be adding the `parse_to` and `retain_raw_logs` attributes to all the plugins that don't currently have them so any new ones should have these fields. This allows the user to choose where to parse.

Here's an [example](https://github.com/observIQ/observiq-otel-collector/blob/release/v1.25.0/plugins/apache_combined_logs.yaml#L17-L27) of them being specified in another plugin.